pkg/cliplugins/kubestellar/syncer-gen/edgesync.go

```diff
@@ -500,24 +499,62 @@ func (o *EdgeSyncOptions) enableSyncerForWorkspace(ctx context.Context, config *
 				OwnerReferences: syncTargetOwnerReferences,
 			},
 			Type: corev1.SecretTypeServiceAccountToken,
-			Data: map[string][]byte{
-				"token": []byte(token.Status.Token),
-			},
 		}
 		tokenSecret, err = kubeClient.CoreV1().Secrets(namespace).Create(ctx, tokenSecret, metav1.CreateOptions{})
 		if err != nil {
 			return "", "", nil, fmt.Errorf("failed to create ServiceAccount token secret: %w", err)
 		}
-	} else {
+		tokenSecretName = tokenSecret.Name
+	}
+	var fetchError error
+	var saTokenBytes []byte
+	err = wait.PollImmediateWithContext(ctx, 300*time.Millisecond, o.ServiceAccountTokenWait, func(ctx context.Context) (bool, error) {
 		tokenSecret, err = kubeClient.CoreV1().Secrets(namespace).Get(ctx, tokenSecretName, metav1.GetOptions{})
-		if err != nil {
-			return "", "", nil, fmt.Errorf("failed to retrieve Secret: %w", err)
+		if err == nil {
+			saTokenBytes = tokenSecret.Data["token"]
+			return len(saTokenBytes) > 0, nil
 		}
+		logger := klog.FromContext(ctx)
+		logger.V(4).Info("Failed to fetch ServiceAccount Secret", "secretName", tokenSecretName, "err", err)
+		fetchError = err
+		return false, nil
+	})
+	if err != nil && err != wait.ErrWaitTimeout {
+		return "", "", nil, fmt.Errorf("failed to get ServiceAccount token, namespace=%v, name=%v, err=%v, fetchError=%v", namespace, tokenSecretName, err, fetchError)
 	}
+	if err == wait.ErrWaitTimeout {
```

Comment:
Is this block required?
I tried syncer-gen on k8s v1.27 (where SA token is not automatically generated) and k8s v1.23 (where SA token is automatically generated). For both cases, the token in placed in `.data.token` of the SA token secret. 
For v1.27, when the SA token secret is manually created at L503 (`kubeClient.CoreV1().Secrets(namespace).Create(ctx, tokenSecret, metav1.CreateOptions{})`), the SA token is also created and populated in `.data.token`.
For v1.23, when SA is created, the SA token secret is automatically created in which the token is populated in `.data.token`.

Reply:
Thank for updating syncer-gen for the campaign in https://github.com/kubestellar/kubestellar/issues/1274.

Reply:
It is safe to introduce this block for just in case that the SA token is successfully populated in the SA token secret.

Reply:
Right. This block covers the context where it currently occurs, using the patched version of kcp v0.11.0 that _does not_ inject an actual token into a service-account-token Secret. Once that is well and truly behind us, I think this block could be removed.